### PR TITLE
Create accessible heading component for Profile, fix a couple edge cases for personal information

### PR DESCRIPTION
**Pull request by adamwhitlock1**
## Description

- Add an accessible heading to the top of each profile page, before the sid nav. This heading is not visible, but is read by the screen readers.

- The section heading is now changed to and H2 and has 'section' added to the heading for screen readers only to differentiate between the section heading and the main H1.

Additional fixes in this PR.

- Added hotfixes for 2 edge cases
  - The gender identity coming back from the api as `null` was invalid for a value in the forms system's radio button select component, so it is set to an empty string instead of `null`
  - The preferred name is now required all the time instead of conditionally, and that way it will prompt a user to add a name or cancel in the event that the user tries to update their name to an empty value.

## Original issue(s)

Main issue for accessibility
https://github.com/department-of-veterans-affairs/va.gov-team/issues/40593

Issue for edge case fixes
https://github.com/department-of-veterans-affairs/va.gov-team/issues/41001

## Testing done
Unit tests added for the new `ProfileAccessibleHeading` component.

E2E tests updated / added to test edge cases

Also tested heading being read first by VoiceOver on mac.

## Screenshots

New dom structure with accessible heading added

Screen Shot 2022-05-09 at 8 32 40 PM

## Acceptance criteria
- [x] Accessible Heading is present and read as the first heading with a screen reader
- [x] Edge cases no longer cause unwanted behaviour

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/personalization/profile/mocks/server.js [hunk 1]
```diff
@@ -27,13 +27,15 @@ const responses = {
...
-      suffix: null,
+    data: {
+      id: '',
+      type: 'hashes',
+      attributes: {
+        first: 'Mitchell',
+        middle: 'G',
+        last: 'Jenkins',
+        suffix: null,
+      },
```
Comment (adamwhitlock1) [inline src/applications/personalization/profile/mocks/server.js:38]: this mock response needed to be nested inside `data` to load correctly. This removes the alert that would display on the profile saying that all info couldn't load fully.

**Review by tmitchellgcio — APPROVED** [review]
LGTM though there are some prop issues that could be done

**Comment by joshkimux** [comment]
> @joshkimux Wanted to loop you in on the accessibility part of this PR and see what your thoughts are on the implementation, and what if any would be a better solution to this.
> 
> We had a discussion with @AngelaFowler82 about the H1 not being the first heading on the Profile pages in this [slack thread](https://dsva.slack.com/archives/C909ZG2BB/p1651153191682419) and this was the idea that I had come up with.

Thanks so much for this @adamwhitlock1 ! This is a thoughtful approach to solving the issue, and I super appreciate the proactiveness at finding an a11y solution for this. This is a11y done right!

That being said, this is also one of those _super_ rare wicked problems, and after talking to Angela I think it's worth putting this on pause until we can loop in @mnorthuis for her perspective on what implications this might have for (1) sitewide IA and (2) future iterations on profile.

Some things we discussed included:
- Potential confusion around similar headings: (1) Personal information vs. (2) Personal information section
- Are the profile links _children_ to the `h1` Personal Information in terms of design intent? Or is it meant as a subnavigation (in which case its current implementation as a secondary navigation is sufficient similar to hub and spoke pages)? Does that meet screen reader user behavior?
- Concerns around future maintenance for using the `sr-only` class as a downstream fix when future solutions come into play (e.g., if we can successfully fix the issue at its root, we'll no longer need the `sr-only` heading)
- Reconciling material dishonesty between the IA, visual design, and DOM is really really hard-- Mikki's perspective is invaluable here

In terms of resolving this, I recommend we:
- Talk to @mnorthuis to get her perspective on this fix
- Discuss what the future of profile may look like and if a "profile" page will ever come to exist as a parent to the subpages that currently are "profile" as a collective
- Weigh a11y concerns against one another and determine what's the more contextually appropriate path forwards

**Comment by adamwhitlock1** [comment]
Going to close this for now, since we are going to continue discussing the accessibility issues and come up with a better game plan for profile.

**Pull request closed without merging**